components/TasksListGroups.js

```diff
@@ -0,0 +1,148 @@
+            task?.dueDate <= thisSunday.toISOString().substring(0, 10)
+        );
+
+  const nextMonday = new Date();
+  nextMonday.setDate(thisSunday.getDate() + 1);
+  const nextTuesday = new Date();
+  nextTuesday.setDate(thisSunday.getDate() + 2);
+  const nextSunday = new Date();
+  nextSunday.setDate(thisSunday.getDate() + 7);
+
+  const nextWeekTasks =
+    today.getDay() === 0
+      ? tasks.filter(
+          (task) =>
+            !task.isDone &&
+            task?.dueDate >= nextTuesday.toISOString().substring(0, 10) &&
+            task?.dueDate <= nextSunday.toISOString().substring(0, 10)
+        )
+      : tasks.filter(
+          (task) =>
+            !task.isDone &&
+            task?.dueDate >= nextMonday.toISOString().substring(0, 10) &&
+            task?.dueDate <= nextSunday.toISOString().substring(0, 10)
+        );
+
+  const afterNextWeekMonday = new Date(nextSunday);
+  afterNextWeekMonday.setDate(afterNextWeekMonday.getDate() + 1);
+  const firstDayNextMonth = new Date(
+    today.getFullYear(),
+    today.getMonth() + 1,
+    1,
+    2
+  );
+
+  const lastDayThisMonth = new Date(firstDayNextMonth - 1);
+
+  const thisMonthTasks = tasks.filter(
+    (task) =>
+      !task.isDone &&
+      task?.dueDate >= afterNextWeekMonday.toISOString().substring(0, 10) &&
+      task?.dueDate <= lastDayThisMonth.toISOString().substring(0, 10)
+  );
+
+  const firstDayAfterNextMonth = new Date(
+    today.getFullYear(),
+    today.getMonth() + 2,
+    1,
+    2
+  );
+
+  const lastDayNextMonth = new Date(firstDayAfterNextMonth - 1);
+  const nextMonthTasks = tasks.filter(
+    (task) =>
+      !task.isDone &&
+      task?.dueDate >= firstDayNextMonth.toISOString().substring(0, 10) &&
+      task?.dueDate <= lastDayNextMonth.toISOString().substring(0, 10)
+  );
+
+  const laterTasks = tasks.filter(
+    (task) =>
+      !task.isDone &&
+      task?.dueDate >= firstDayAfterNextMonth.toISOString().substring(0, 10)
+  );
+
+  const completedTasks = tasks.filter((task) => task.isDone);
+
+  const tasksGroupsData = [
+    { tasks: missedTasks, groupKey: "Missed", red: true },
+    { tasks: todaysTasks, groupKey: "Today" },
+    { tasks: tomorrowsTasks, groupKey: "Tomorrow" },
+    { tasks: thisWeekTasks, groupKey: "This week" },
+    { tasks: nextWeekTasks, groupKey: "Next week" },
+    { tasks: thisMonthTasks, groupKey: "This month" },
+    { tasks: nextMonthTasks, groupKey: "Next month" },
+    { tasks: laterTasks, groupKey: "Later" },
+    { tasks: completedTasks, groupKey: "Completed" },
+  ];
+  return (
+    <>
+      {tasksGroupsData.map(
+        (taskGroupData) =>
+          taskGroupData.tasks.length > 0 && (
+            <TasksListGroup
+              key={taskGroupData.groupKey}
+              tasks={taskGroupData.tasks}
+              setDetailsBackLinkRef={setDetailsBackLinkRef}
+              groupKey={taskGroupData.groupKey}
+              groupTitle={taskGroupData.groupKey}
```

Comment:
You are passing the same thing here. I would suggest just passing the `groupKey` once as a prop and using it in your component for both, the key and title.
```suggestion
              groupKey={taskGroupData.groupKey}
```